### PR TITLE
Revert "rust: catch exceptions in the entire execute() wrapper"

### PR DESCRIPTION
**Pull request by axic**
Reverts #393.

### CODE REVIEW

**Comment by axic** [comment]
> With the additional commit, this much more than revert.

Yes, it is better than a revert. Achieves the original goal :wink:

File: bindings/rust/evmc-declare/src/lib.rs [hunk 1]
```diff
@@ -337,40 +341,40 @@ fn build_execute_fn(names: &VMNameSet) -> proc_macro2::TokenStream {
...
+            // TODO: context is optional in case of the "precompiles" capability
+            if instance.is_null() || context.is_null() || msg.is_null() || (code.is_null() && code_size != 0) {
+                // These are irrecoverable errors that violate the EVMC spec.
+                std::process::abort();
+            }
 
-                let execution_message: ::evmc_vm::ExecutionMessage = unsafe {
-                    msg.as_ref().expect("EVMC message is null").into()
-                };
+            assert!(!instance.is_null());
```
Comment (jakelang) [inline bindings/rust/evmc-declare/src/lib.rs]: The above conditional should handle all three of these asserts.

File: bindings/rust/evmc-declare/src/lib.rs [hunk 2]
```diff
@@ -337,40 +341,40 @@ fn build_execute_fn(names: &VMNameSet) -> proc_macro2::TokenStream {
...
+            let execution_message: ::evmc_vm::ExecutionMessage = unsafe {
+                msg.as_ref().expect("EVMC message is null").into()
+            };
 
-                let container = unsafe {
-                    ::evmc_vm::EvmcContainer::<#type_name_ident>::from_ffi_pointer(instance)
-                };
+            let empty_code = [0u8;0];
+            let code_ref: &[u8] = if code.is_null() {
+                assert_eq!(code_size, 0);
```
Comment (jakelang) [inline bindings/rust/evmc-declare/src/lib.rs]: Also handled by the first conditional.

File: bindings/rust/evmc-declare/src/lib.rs (hunk 1 above)
Comment (axic) [reply bindings/rust/evmc-declare/src/lib.rs]: I know, though I saw no harm keeping the asserts.

**Pull request merged**